### ISSUE TITLE
Inactivity Tracking Service requires Lavalink4NET logger (though it should be optional)

### ISSUE THREAD

**Issue by Mijago** [angelobreuer/Lavalink4NET#6]
**Describe the bug**
Inactivity Tracking Service requires Lavalink4NET logger (though it should be optional).
I registered the following code:
```
                .AddSingleton<InactivityTrackingOptions>()
                .AddSingleton<InactivityTrackingService>()
```
And later used:
```
            _provider.GetRequiredService<InactivityTrackingService>().BeginTracking();
```

Then I get this nice error message:
```
Unhandled Exception: System.InvalidOperationException: Unable to resolve service for type 'Lavalink4NET.ILogger' while attempting to activate 'Lavalink4NET.Tracking.InactivityTrackingService'.
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateArgumentCallSites(Type serviceType, Type implementationType, CallSiteChain callSiteChain, ParameterInfo[] parameters, Boolean throwIfCallSiteNotFound)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateConstructorCallSite(Type serviceType, Type implementationType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(ServiceDescriptor descriptor, Type serviceType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(Type serviceType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateCallSite(Type serviceType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.CreateServiceAccessor(Type serviceType)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at MusicBot.Services.StartupService.InitializeLavalink() in /app/Services/StartupService.cs:line 69
   at MusicBot.Services.StartupService.StartAsync() in /app/Services/StartupService.cs:line 64
   at MusicBot.Startup.RunAsync() in /app/Startup.cs:line 47
   at MusicBot.Startup.RunAsync(String[] args) in /app/Startup.cs:line 31
   at MusicBot.Program.<Main>(String[] args)
```

**To Reproduce**
- Use an InactivityTrackingService without registering an Lavalink4NET.ILogger.

**Expected behavior**
It should run without requiring a logger.

**(please complete the following information):**
 - Lavalink4NET Version 1.2.0
 - Exception thrown? (consider enabling logging)

**Additional context**
I'll just solve the issue on my side by adding a Lavalink4NET.ILogger, but the wiki states that logging is optional, so I just wanted to let you know that here may exist an issue.

**Comment by Mijago** [comment]
This gives me another question:
I add my ILogger to my provider by using `.AddLogging(builder => builder.AddConsole())` (for testing etc.).

Then I register Lavalink-related stuff using:
```
.AddSingleton<IAudioService, LavalinkNode>()
.AddSingleton<IDiscordClientWrapper, DiscordClientWrapper>()
.AddSingleton(new LavalinkNodeOptions
{
    RestUri = "http://lavalink:8080/",
    WebSocketUri = "ws://lavalink:8080/",
    Password = "youshallnotpass",
    AllowResuming = false
})
.AddSingleton<InactivityTrackingOptions>() // This is new
.AddSingleton<InactivityTrackingService>() // This is new, too
```
So my question is: How do I get the logger to the LavalinkNode instance when using DI? 


Offtopic:
By the way, good work on this library! I enjoy working with it.

**Comment by angelobreuer** [comment]
The problem is with Dependency Injection, for example when creating the instance of InactivityTrackingService using its constructor the issue is solved, I added to the 1.3.0-release milestone adding service provider constructors (or constructors without the optional parameters) for use with dependency injection. 

As a "work-around" you could use the constructor of the InactivityTrackingService to avoid the ILogger as shown below:

```csharp
new ServiceCollection()
    // [...] adding client wrapper, etc.
    .AddSingleton(sp => new InactivityTrackingService(sp.GetRequiredService<IAudioService>(),
        sp.GetRequiredService<IDiscordClientWrapper>(),
        sp.GetRequiredService<InactivityTrackingOptions>(),
        null))
    // [...]
    .BuildServiceProvider();
```

**Comment by angelobreuer** [comment]
> This gives me another question:
> I add my ILogger to my provider by using `.AddLogging(builder => builder.AddConsole())` (for testing etc.).
> 
> Then I register Lavalink-related stuff using:
> 
> ```
> .AddSingleton<IAudioService, LavalinkNode>()
> .AddSingleton<IDiscordClientWrapper, DiscordClientWrapper>()
> .AddSingleton(new LavalinkNodeOptions
> {
>     RestUri = "http://lavalink:8080/",
>     WebSocketUri = "ws://lavalink:8080/",
>     Password = "youshallnotpass",
>     AllowResuming = false // (btw, I can pause/resume despite this setting)
> })
> .AddSingleton<InactivityTrackingOptions>() // This is new
> .AddSingleton<InactivityTrackingService>() // This is new, too
> ```
> 
> So my question is: How do I get the logger to the LavalinkNode instance when using DI?
> 
> Offtopic:
> By the way, good work on this library! I enjoy working with it.

The ILogger-interface used by Lavalink4NET is not from Microsoft.Extensions.Logging(.Abstractions) anymore (since version 1.2.0, to remove the dependency).

See: https://github.com/angelobreuer/Lavalink4NET/blob/master/samples/Lavalink4NET.Discord_NET.ExampleBot/CustomLogger.cs for an example implementation of the interface. (It must be also added to the dependency injection service provider)

**Comment by angelobreuer** [comment]
Hope the issue is fixed now with #8, release version 1.3.0 is now available (also on NuGet).
